src/Parser/Shaman/Elemental/Modules/ShamanCore/EarthShock.js

```diff
@@ -0,0 +1,126 @@
+        if (this.combatants.selected.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id)) {
+          this.maelstromSpendInAscendence.push(event.classResources[0].amount);
+        }
+        else {
+          this.maelstromSpendOutAscendence.push(event.classResources[0].amount);
+        }
+      }
+    }
+    if (event.classResources[0].amount < 40 && event.classResources[0].type === RESOURCE_TYPES.MAELSTROM.id) {
+      this.latershock = false;
+    }
+  }
+
+  suggestions(when) {
+    if (this.gambling) {
+      when(Math.abs((70 + 85) / 2 - this.avgMaelstromSpendOutOfAsc)).isGreaterThan(12.5)
+        .addSuggestion((suggest) => {
+          return suggest(<span>Try to cast <SpellLink id={SPELLS.EARTH_SHOCK.id} /> at 70 to 85 Maelstrom when using
+          <ItemLink id={ITEMS.SMOLDERING_HEART.id} /> and <ItemLink id={ITEMS.THE_DECEIVERS_BLOOD_PACT.id} /> together.
+          <SpellLink id={SPELLS.AFTERSHOCK_TALENT.id} /> causes procs from <ItemLink id={ITEMS.THE_DECEIVERS_BLOOD_PACT.id} />  to <em>generate</em> Maelstrom.
+          Casting at 70 to 85 Maelstrom allows for a buffer to avoid Maelstrom overflow on chained <ItemLink id={ITEMS.THE_DECEIVERS_BLOOD_PACT.id} />  procs.</span>)
+            .icon(SPELLS.EARTH_SHOCK.icon)
+            .actual(`Average Maelstrom spent: ${formatNumber(this.avgMaelstromSpendOutOfAsc)}`)
+            .recommended(`70 to 85 is recommended`)
+            .regular(5).major(15);
+        });
+    }
+    else if (this.pseudogambling) {
+      when(Math.abs((85 + 95) / 2 - this.avgMaelstromSpendOutOfAsc)).isGreaterThan(12.5)
+        .addSuggestion((suggest) => {
+          return suggest(<span>Try to cast <SpellLink id={SPELLS.EARTH_SHOCK.id} /> at 85 to 95 Maelstrom when using
+          <ItemLink id={ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id} /> and <ItemLink id={ITEMS.THE_DECEIVERS_BLOOD_PACT.id} /> together.
+          <SpellLink id={SPELLS.AFTERSHOCK_TALENT.id} /> causes procs from <ItemLink id={ITEMS.THE_DECEIVERS_BLOOD_PACT.id} />  to <em>generate</em> Maelstrom.
+          Casting at 85 to 95 Maelstrom allows for a buffer to avoid Maelstrom overflow on chained <ItemLink id={ITEMS.THE_DECEIVERS_BLOOD_PACT.id} />  procs.</span>)
+            .icon(SPELLS.EARTH_SHOCK.icon)
+            .actual(`Average Maelstrom spent: ${formatNumber(this.avgMaelstromSpendOutOfAsc)}`)
+            .recommended(`85 to 95 is recommended`)
+            .regular(5).major(15);
+        });
+    }
+    else {
+      when(this.avgMaelstromSpendOutOfAsc).isLessThan(111)
+        .addSuggestion((suggest) => {
+          return suggest(<span>Try to cast <SpellLink id={SPELLS.EARTH_SHOCK.id} /> at 111 Maelstrom or more.</span>)
+            .icon(SPELLS.EARTH_SHOCK.icon)
+            .actual(`Average Maelstrom spent: ${formatNumber(this.avgMaelstromSpendOutOfAsc)}`)
+            .recommended(`More than 111 is recommended`)
+            .regular(5).major(15);
+        });
+    }
+
+    when(this.avgMaelstromSpendInAsc).isLessThan(111)
```

Comment:
This is duplicate of the suggestion above - is that a typo or do you *really* want to have 2 identical suggestions when both `gambling` and `pseudogambling` are `false`?

Reply:
This one checks for spending outside of Ascendence.

So it should be displayed independently from the other ones and regardless of gambling or not

Reply:
Oh, my bad, I didn't see the in/out. Still, the suggestion for this whole block is correct even for this block.